### PR TITLE
Use of appsettings for track2 samples, update to .Net 7

### PR DESCRIPTION
**Pull request by xpouyat**
## Purpose

* Migration of all samples to .Net 7
* Use of appsettings.json for track2 samples
* Tested with VS Code
* Removed of .env file for track2 samples
* Migration to AMS SDK 1.1

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[X] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: Account/Quotas/Program.cs [hunk 1]
```diff
@@ -4,52 +4,113 @@
 using Azure.Identity;
 using Azure.Monitor.Query;
 using Azure.ResourceManager.Media;
+using Microsoft.Extensions.Configuration;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
 
-var quotas = new QuotaMetrics[]
+internal class Program
```
Comment (jonpayne) [inline Account/Quotas/Program.cs:11]: Why is this being wrapped in a class?

File: Assets/Assets.csproj [hunk 2]
```diff
@@ -1,19 +1,25 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
-  <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
-  </PropertyGroup>
+	<PropertyGroup>
```
Comment (jonpayne) [inline Assets/Assets.csproj:3]: It looks like these indents have changed from spaces to tabs. I think we should use spaces everywhere.

File: Assets/Program.cs [hunk 3]
```diff
@@ -5,46 +5,110 @@
...
+using Microsoft.Extensions.Configuration;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 
-var MediaServiceAccount = MediaServicesAccountResource.CreateResourceIdentifier(
-    subscriptionId: "---set-your-subscription-id-here---",
-    resourceGroupName: "---set-your-resource-group-name-here---",
-    accountName: "---set-your-media-services-account-name-here---");
+namespace Assets
```
Comment (jonpayne) [inline Assets/Program.cs:13]: I would like to avoid namespaces and default classes where they are not needed.

File: ConsoleApp1/ConsoleApp1.csproj [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (jonpayne) [inline ConsoleApp1/ConsoleApp1.csproj:1]: Was this file intended to be in this PR?

File: ConsoleApp2/ConsoleApp2.csproj [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (jonpayne) [inline ConsoleApp2/ConsoleApp2.csproj:1]: Was this file intended to be in this PR?

File: ContentProtection/OfflinePlayReadyAndWidevine/OfflinePlayReadyAndWidevine.csproj [hunk 6]
```diff
@@ -1,37 +1,36 @@
...
+		<OutputType>Exe</OutputType>
+		<TargetFramework>net7.0</TargetFramework>
+		<ImplicitUsings>enable</ImplicitUsings>
+		<Nullable>enable</Nullable>
+	</PropertyGroup>
 
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <LangVersion>latest</LangVersion>
-  </PropertyGroup>
+	<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (jonpayne) [inline ContentProtection/OfflinePlayReadyAndWidevine/OfflinePlayReadyAndWidevine.csproj:10]: This property group will likely break release builds.

File: ContentProtection/OfflineFairPlay/Program.cs [hunk 7]
```diff
@@ -345,7 +345,7 @@ private static async Task RunAsync(ConfigWrapper config)
         /// <returns></returns>
         private static async Task<Asset> CreateOutputAssetAsync(IAzureMediaServicesClient client, string resourceGroupName, string accountName, string assetName)
         {
-            Asset outputAsset = new Asset();
+            Asset outputAsset = new();
```
Comment (jonpayne) [inline ContentProtection/OfflineFairPlay/Program.cs:348]: I'm not keen on this style. I think we should be consistent and use `var` everywhere.

File: Streaming/StreamHLSAndDASH/StreamHLSAndDASH.csproj [hunk 8]
```diff
@@ -1,39 +1,38 @@
...
-    </None>
-  </ItemGroup>
+	<PropertyGroup>
+		<OutputType>Exe</OutputType>
+		<TargetFramework>net7.0</TargetFramework>
+		<ImplicitUsings>enable</ImplicitUsings>
+		<Nullable>enable</Nullable>
+	</PropertyGroup>
+
+	<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (jonpayne) [inline Streaming/StreamHLSAndDASH/StreamHLSAndDASH.csproj:10]: This will likely break release builds.

File: ConsoleApp1/ConsoleApp1.csproj (hunk 4 above)
Comment (xpouyat) [reply ConsoleApp1/ConsoleApp1.csproj]: You are right. It is removed.

File: ConsoleApp2/ConsoleApp2.csproj (hunk 5 above)
Comment (xpouyat) [reply ConsoleApp2/ConsoleApp2.csproj]: Removed

File: ContentProtection/OfflineFairPlay/Program.cs (hunk 7 above)
Comment (xpouyat) [reply ContentProtection/OfflineFairPlay/Program.cs:348]: I liked it :-)

File: Account/Quotas/Program.cs (hunk 1 above)
Comment (xpouyat) [reply Account/Quotas/Program.cs:11]: I find it clearer that sample code without a class.

Comment (jonpayne) [reply Account/Quotas/Program.cs:11]: For these simple sample applications, wrapping the code in a class (and sometimes a namespace), doesn't add any value.

File: ContentProtection/OfflinePlayReadyAndWidevine/OfflinePlayReadyAndWidevine.csproj (hunk 6 above)
Comment (xpouyat) [reply ContentProtection/OfflinePlayReadyAndWidevine/OfflinePlayReadyAndWidevine.csproj:10]: I don't understand why it would break release builds. I just tested. Changes are just a formatting change and this two new lines
<ImplicitUsings>enable</ImplicitUsings>
<Nullable>enable</Nullable>

File: Streaming/StreamHLSAndDASH/StreamHLSAndDASH.csproj (hunk 8 above)
Comment (jonpayne) [reply Streaming/StreamHLSAndDASH/StreamHLSAndDASH.csproj:10]: It is only setting LangVersion to "latest" for Debug builds. LangVersion should be set to the same value for all build types.

File: Account/Quotas/Program.cs (hunk 1 above)
Comment (xpouyat) [reply Account/Quotas/Program.cs]: Ok, I removed them

File: ContentProtection/OfflineFairPlay/Program.cs (hunk 7 above)
Comment (xpouyat) [reply ContentProtection/OfflineFairPlay/Program.cs]: Fixed

File: Assets/Assets.csproj (hunk 2 above)
Comment (xpouyat) [reply Assets/Assets.csproj]: Fixed

File: ContentProtection/OfflinePlayReadyAndWidevine/OfflinePlayReadyAndWidevine.csproj (hunk 6 above)
Comment (xpouyat) [reply ContentProtection/OfflinePlayReadyAndWidevine/OfflinePlayReadyAndWidevine.csproj]: Removed

File: Streaming/StreamHLSAndDASH/StreamHLSAndDASH.csproj (hunk 8 above)
Comment (xpouyat) [reply Streaming/StreamHLSAndDASH/StreamHLSAndDASH.csproj]: Removed
